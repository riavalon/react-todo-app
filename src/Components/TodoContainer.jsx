import React from 'react'
import Todo from './Todo'
import styles from '../Styles/TodoContainer.module.css'

const TodoContainer = props => {
  return (
    <ul className={styles.list}>
      {props.todos.map(item => {
        return (
          <Todo 
            key={item.key}
            id={item.key}
            text={item.text}
            onClick={() => props.onClick(item.key)}
          />
        )
      })}
      {
        props.todos.length === 0 && (
          <li className={styles.emptyList}>
            <p>No outstanding tasks! <span role="img" className={styles.emojis} aria-label="emojis">ðŸŽ‰ðŸŽ‰</span></p>
          </li>
        )
      }
    </ul>
  )
}
export default TodoContainer