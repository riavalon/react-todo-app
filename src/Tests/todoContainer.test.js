import React from 'react';
import renderer from 'react-test-renderer';
import TodoContainer from '../Components/TodoContainer';

it('renders correctly', () => {
  const todos = [
    {
      text: 'Laundry',
      key: 1,
    },
    {
      text: 'Grocery Shopping',
      key: 2,
    },
  ]
  const tree = renderer.create(
    <TodoContainer todos={todos} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});