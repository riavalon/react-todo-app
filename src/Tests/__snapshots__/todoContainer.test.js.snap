// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="list"
>
  <li
    className="listItem"
    onClick={[Function]}
  >
    Laundry
  </li>
  <li
    className="listItem"
    onClick={[Function]}
  >
    Grocery Shopping
  </li>
</ul>
`;
